osu.Game.Rulesets.Catch/UI/DrawableCatchRuleset.cs

```diff
@@ -52,5 +53,7 @@ private void load()
         protected override PassThroughInputManager CreateInputManager() => new CatchInputManager(Ruleset.RulesetInfo);
 
         public override DrawableHitObject<CatchHitObject>? CreateDrawableRepresentation(CatchHitObject h) => null;
+
+        protected override ResumeOverlay CreateResumeOverlay() => new DelayedResumeOverlay();
```

Comment:
This doesn't actually work on its own. This is because catch "provides a cursor":

https://github.com/ppy/osu/blob/0ebea77392fbca045611e8478494861b0919ef04/osu.Game.Rulesets.Catch/UI/CatchCursorContainer.cs#L9-L14

triggering this logic:

https://github.com/ppy/osu/blob/0ebea77392fbca045611e8478494861b0919ef04/osu.Game/Rulesets/UI/DrawableRuleset.cs#L230

The ad-hoc solution is to do this:

```diff
diff --git a/osu.Game.Rulesets.Catch/UI/CatchCursorContainer.cs b/osu.Game.Rulesets.Catch/UI/CatchCursorContainer.cs
index 4ae61ef8c7..fda5659769 100644
--- a/osu.Game.Rulesets.Catch/UI/CatchCursorContainer.cs
+++ b/osu.Game.Rulesets.Catch/UI/CatchCursorContainer.cs
@@ -11,5 +11,11 @@ public partial class CatchCursorContainer : GameplayCursorContainer
         // Just hide the cursor.
         // The main goal here is to show that we have a cursor so the game never shows the global one.
         protected override Drawable CreateCursor() => Empty();
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            Show();
+        }
     }
 }

```

but I'm not sure how to feel about that.

Reply:
Should be fixed, but I took a different route of moving the cursor related logic into `OsuResumeOverlay`.